File: anagrams.rb
# http://rosettacode.org/wiki/Anagrams

def find_anagrams(words)
  anagrams = Hash.new {|hash,key| hash[key] = []}
  longest_anagrams = []
  words.each do |word|
    anagrams[word.split(//).sort] << word
  end
  max_count = anagrams.values.map {|anagram| anagram.length}.max
  anagrams.each_value do |anagram|
    longest_anagrams << anagram if anagram.count >= max_count
  end
  longest_anagrams
end

def unix_words
  words_to_test = File.open("unixdict.txt").readlines
  words_to_test.map {|word| word.chomp}
end

find_anagrams(unix_words).each do |anagrams|
  p anagrams
end    

File: anagrams.swift
// http://rosettacode.org/wiki/Anagrams

import Foundation

func findAnagrams(file: String) ->  [String:[String]] {
  let fileContent = NSString(contentsOfFile: file, encoding: NSUTF8StringEncoding, error: nil)!
  let words = fileContent.componentsSeparatedByString("\n") as! [String]
  var anagramDict = [String:[String]]()
  for word in words {
    let chars = splitWordIntoLetters(word as String)
    if anagramDict[chars] == nil {
      anagramDict[chars] = [word as String]
    } else {
      var values = anagramDict[chars]
      values?.append(word)
      anagramDict[chars] = values 
    }
  }
  return anagramDict
}

func splitWordIntoLetters(word: String) -> String {
  let char = Array(word).sorted() { $0 < $1 }
  let stringResult = char.description
  return stringResult
}

func countAnagrams() {
  let anagrams = findAnagrams("unixdict.txt")
  var anagramCount = 4
  var mostAnagrams: [String] = []
  for (key, value) in anagrams {
    if value.count >= anagramCount {
      println("\(value.count): \(value)")
    }
  }
}

countAnagrams()
